health-sdk/sdk/src/main/java/net/gini/android/health/sdk/review/ReviewFragment.kt

```diff
@@ -547,25 +554,34 @@ class ReviewFragment private constructor(
     private fun showOpenWithDialog(paymentProviderApp: PaymentProviderApp) {
         OpenWithBottomSheet.newInstance(paymentProviderApp, object: OpenWithForwardListener {
             override fun onForwardSelected() {
-                startSharePdfIntent()
+                viewModel.onForwardToSharePdfTapped()
             }
         }).also {
             it.show(requireActivity().supportFragmentManager, it::class.java.name)
         }
         viewModel.incrementOpenWithCounter()
     }
 
-    private fun startSharePdfIntent() {
-        //TODO link with downloaded PDF file after backend is ready
+    private fun startSharePdfIntent(paymentRequestFile: File) {
+        val uriForFile = FileProvider.getUriForFile(
+            requireContext(),
+            requireContext().packageName+".health.sdk.fileprovider",
+            paymentRequestFile
+        )
         val shareIntent = Intent(Intent.ACTION_SEND).apply {
             type = "application/pdf"
+            flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+            putExtra(Intent.EXTRA_STREAM, uriForFile)
         }
-        startActivity(Intent.createChooser(shareIntent, ""))
+        startActivity(Intent.createChooser(shareIntent, uriForFile.lastPathSegment))
     }
 
     private fun handlePaymentNextStep(paymentNextStep: ReviewViewModel.PaymentNextStep) {
         when (paymentNextStep) {
-            ReviewViewModel.PaymentNextStep.OpenSharePdf -> startSharePdfIntent()
+            ReviewViewModel.PaymentNextStep.DownloadPaymentRequestFile -> {
+                binding.loading.isVisible = true
+                viewModel.getFileAsByteArray(requireContext().externalCacheDir)
```

Comment:
[Suggestion] I think we are mixing the responsibilities of the fragment and the viewmodel. Currently the code flow for getting the pdf is (ignoring the "open with" onboarding sheet for simplicity):
1. Fragment: notify ViewModel that payment button was tapped
2. ViewModel: notify Fragment to download the pdf
3. Fragment: notify ViewModel to download the pdf
4. ViewModel: notify Fragment to share the pdf

Could the flow be simplified to:
1. Fragment: notify ViewModel that payment button was tapped
2. ViewModel: 
  a. notify Fragment to show loading indicator for pdf download
  b. download pdf
  c. notify Fragment to hide loading indicator for pdf download
  d. notify Fragment to share the pdf OR to show error

Reply:
Yes, I was overcomplicating things a bit...simplified the flow, but sending `externalCacheDir` as a param when tapping on payment button.